Java/other/AggressiveCows.java

```diff
@@ -0,0 +1,81 @@
+import java.util.*;
+
+public class AggressiveCow {
+
+    private static boolean checkDist(int dist, int cows, int arr[]) {
+        int cowInStall = 1;
+        int lastPos = arr[0];
+
+        // here we check if the distance we have can be minimum distance or not
+        // if yes, we return true
+        for (int i = 1; i < arr.length; i++) {
+            if (arr[i] - lastPos < dist) {
+                continue;
+            }
+            if(++cowInStall == cows) {
+                return true;
+            }
+            lastPos = arr[i];
+        }
+
+        return false;
+    }
+
+    private static int binSearch(int cows, int arr[]) {
+        int left = 0;
+        int right = arr[arr.length - 1] - arr[0];
+
+        // using binary search technique to find distances
+        while (left <= right) {
+            int mid = (left + right)/2;
```

Comment:
@ritvij14 use `mid = left + ((right - left) / 2)` this will avoid overflow conditon.

Reply:
@iamrajiv done.